src/sql/workbench/parts/notebook/models/cell.ts

```diff
@@ -382,11 +389,39 @@ export class CellModel implements ICellModel {
 		if (output) {
 			// deletes transient node in the serialized JSON
 			delete output['transient'];
+			this.checkFailure(output);
 			this._outputs.push(this.rewriteOutputUrls(output));
 			this.fireOutputsChanged();
 		}
 	}
 
+	private async checkFailure(output: nb.ICellOutput): Promise<void> {
```

Comment:
This isn't the right place to handle this. Can can/should put this in the Jupyter kernel logic and be checking there before runing the code. That's why this is failing. We're attempting to do things too late, and also this special-cases 401 errors when that's not a scalable solution.